### PR TITLE
create group not anchor with restaurant

### PR DESCRIPTION
**Pull request by ducbet**

### CODE REVIEW

File: app/src/main/java/com/framgia/feastival/util/FeastivalApplication.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Created by tmd on 12/08/2017.
+ */
+public class FeastivalApplication extends Application {
+    private static FeastivalApplication mInstance;
+
+    @Override
+    public void onCreate() {
+        mInstance = this;
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/util/FeastivalApplication.java]: ERROR: [STYLE][ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD] Write to static field com.framgia.feastival.util.FeastivalApplication.mInstance from instance method com.framgia.feastival.util.FeastivalApplication.onCreate()
http://findbugs.sourceforge.net/bugDescriptions.html#ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD

File: app/src/main/java/com/framgia/feastival/util/FeastivalApplication.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import android.content.Context;
+
+/**
+ * Created by tmd on 12/08/2017.
+ */
+public class FeastivalApplication extends Application {
+    private static FeastivalApplication mInstance;
+
+    @Override
+    public void onCreate() {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/util/FeastivalApplication.java]: WARNING: [CallSuperFirst] super should be called at the start of the method
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/android.html#CallSuperFirst

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -44,4 +44,5 @@
     <string name="create_new_group">Create new group</string>
     <string name="create_new_restaurant">Create new restaurant</string>
     <string name="add_more_view_point">Add more view point</string>
+    <string name="double_click_back">Please click BACK again to exit</string>
```
Comment (DoanVanToan) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.double_click_back` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -45,4 +45,5 @@
     <string name="create_new_restaurant">Create new restaurant</string>
     <string name="add_more_view_point">Add more view point</string>
     <string name="creating_new_group_state">You are creating new group state</string>
+    <string name="double_click_back">Please click BACK again to exit</string>
```
Comment (DoanVanToan) [inline app/src/main/res/values/strings.xml]: WARNING: [UnusedResources] The resource `R.string.double_click_back` appears to be unused
 Unused resources make applications larger and slow down builds.

File: app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java [hunk 5]
```diff
@@ -398,28 +425,69 @@ public void setSelectedRestaurant(Marker marker) {
...
+
+    private String getAddress(LatLng latLng) {
+        Geocoder gcd = new Geocoder(mContext, Locale.getDefault());
+        List<Address> addresses = null;
+        try {
+            addresses = gcd.getFromLocation(latLng.latitude, latLng.longitude, 1);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        if (addresses.size() > 0)
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java]: WARNING: 'if' construct must use '{}'s.

File: app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java [hunk 6]
```diff
@@ -398,28 +425,69 @@ public void setSelectedRestaurant(Marker marker) {
...
+        }
+    }
+
+    private String getAddress(LatLng latLng) {
+        Geocoder gcd = new Geocoder(mContext, Locale.getDefault());
+        List<Address> addresses = null;
+        try {
+            addresses = gcd.getFromLocation(latLng.latitude, latLng.longitude, 1);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java]: WARNING: [AvoidPrintStackTrace] Avoid printStackTrace(); use a logger call instead.
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#AvoidPrintStackTrace

File: app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java (hunk 5 above)
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java]: WARNING: [IfStmtsMustUseBraces] Avoid using if statements without curly braces
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/braces.html#IfStmtsMustUseBraces

File: app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java [hunk 7]
```diff
@@ -398,28 +425,69 @@ public void setSelectedRestaurant(Marker marker) {
...
+        }
+        if (addresses.size() > 0)
+            System.out.println(addresses.get(0).getLocality());
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainViewModel.java]: ERROR: [SystemPrintln] System.out.println is used
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/logging-java.html#SystemPrintln

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (ducbet) [inline app/src/main/res/values/strings.xml]: Em dùng trong MainActivity.java mà nó vẫn bị warning ạ

File: app/src/main/java/com/framgia/feastival/screen/main/MainActivity.java [hunk 8]
```diff
@@ -114,4 +116,20 @@ protected void onStop() {
...
+            return;
+        }
+        mIsDoubleClickedBack = true;
+        Toast.makeText(this, getString(R.string.double_click_back), Toast.LENGTH_LONG).show();
+        new Handler().postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                mIsDoubleClickedBack = false;
+            }
+        }, 2000);
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/feastival/screen/main/MainActivity.java]: pls define final

**Pull request merged**